src/mscorlib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -537,7 +538,7 @@ internal String ToString(TraceFormat traceFormat)
             {
                 StackFrame sf = GetFrame(iFrameIndex);
                 MethodBase mb = sf.GetMethod();
-                if (mb != null)
+                if (mb != null && !(mb.IsDefined(typeof(StackTraceHiddenAttribute)) || mb.DeclaringType.IsDefined(typeof(StackTraceHiddenAttribute))))
```

Comment:
Could you factor out the new part of the conditional into a little helper method and make it clear with the naming/comments that this is our filtering policy? I suspect there may be some continued tweaking over time and I want to keep the policy choice isolated from the code here that applies the policy.

Reply:
Done